framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4631,6 +4619,452 @@ private void makeGroundTargetType(
+      AnnotatedTypeMirror capturedTypeArg = capturedType.getTypeArguments().get(i);
+      if (uncapturedTypeArg.getKind() == TypeKind.WILDCARD) {
+        if (TypesUtils.isCapturedTypeVariable(capturedTypeArg.getUnderlyingType())) {
+          // The type argument is a captured type variable. Use the type argument from the newly
+          // created and yet-to-be annotated capturedType. (The annotations are added as
+          // part of capturing the wildcard.)
+          typeVariableMapping.put(typeVarTypeMirror, capturedTypeArg);
+          // Also, add a mapping from the captured type variable to the annotated captured
+          // type variable, so that if one captured type variable refers to another, the same
+          // AnnotatedTypeVariable object is used.
+          capturedTypeVarMapping.put(
+              ((AnnotatedTypeVariable) capturedTypeArg).getUnderlyingType(),
+              (AnnotatedTypeVariable) capturedTypeArg);
+          newTypeArgs.add(capturedTypeArg);
+        } else {
+          // Javac used a declared type instead of a captured type variable.  This seems to happen
+          // when the bounds of the captured type variable would have been identical. This seems to
+          // be a violation of the JLS, but javac does this, so the Checker Framework handle that
+          // case.
+          typeVariableMapping.put(typeVarTypeMirror, capturedTypeArg);
+          newTypeArgs.add(capturedTypeArg);
+          replaceAnnotations(
+              ((AnnotatedWildcardType) uncapturedTypeArg).getSuperBound(), capturedTypeArg);
+        }
+      } else {
+        // The type argument is not a wildcard.
+        typeVariableMapping.put(typeVarTypeMirror, uncapturedTypeArg);
+        if (uncapturedTypeArg.getKind() == TypeKind.TYPEVAR) {
+          AnnotatedTypeVariable typeVar = (AnnotatedTypeVariable) uncapturedTypeArg;
+          typeVariableMapping.put(typeVar.getUnderlyingType(), typeVar);
+        }
+        newTypeArgs.add(uncapturedTypeArg);
+      }
+    }
+
+    // Loop through the type arguments and set the annotations of each captured type variable.
+    List<AnnotatedTypeVariable> orderToCapture = order(capturedTypeVarMapping.values());
+    for (AnnotatedTypeVariable capturedTypeArg : orderToCapture) {
+      int i = capturedTypeMirror.getTypeArguments().indexOf(capturedTypeArg.getUnderlyingType());
+      AnnotatedTypeMirror uncapturedTypeArg = uncapturedType.getTypeArguments().get(i);
+      AnnotatedTypeVariable typeVariable =
+          (AnnotatedTypeVariable) typeDeclaration.getTypeArguments().get(i);
+      captureWildcard(
+          typeVariableMapping,
+          capturedTypeVarMapping,
+          (AnnotatedWildcardType) uncapturedTypeArg,
+          typeVariable,
+          capturedTypeArg);
+      newTypeArgs.remove(i);
+      newTypeArgs.add(i, capturedTypeArg);
+    }
+
+    capturedType.setTypeArguments(newTypeArgs);
+    // Use the mapping above to substitute the type variables in capturedType.
```

Comment:
This is a confusing location for a comment.  If it refers to the line above, put it before that line.  If it refers to the line below, remove the separating whitespace and maybe add whitespace before the comment.